pkg/network/ovn_kubernetes.go

```diff
@@ -2420,8 +2420,6 @@ func prepareUpgradeToInterConnect(ovn bootstrap.OVNBootstrapResult, client cnocl
 		!ovn.NodeUpdateStatus.InterConnectEnabled &&
 		!ovn.MasterUpdateStatus.InterConnectEnabled &&
 		doesVersionEnableInterConnect(os.Getenv("RELEASE_VERSION")) &&
-		!ovn.NodeUpdateStatus.Progressing &&
```

Comment:
so in this function, we have 3 if conditions:
1. Is phase 1 about start
2. Is phase 2 about to start
3. Is phase 2 complete

Should we also have an else here, that prints the current status, like "Upgrade in progress to phase 1". I'm not sure how often this function is invoked, but seeing some periodic status might help. Or maybe additionally/alternatively would it be good to have an else clause where we get to some condition we never expect to happen. Like skipping phase 1 and log an error?

Reply:
Yeah, in this function I start and end each phase, then in `renderOVNKubernetes` the code is not aware of IC upgrades: it knows what to do when switching from single zone to multi zone (and back) and it knows how to do updates (as it did before, nothing new there).

I've just added a few lines in `prepareUpgradeToInterConnect` that should print at each iteration of CNO where it stands w.r.t the two phases in the IC upgrade. I'll validate that with a few upgrade jobs.

Reply:
So with these changes, a simple loki query would unveil how CNO proceeded with the upgrade and when:
(Query: `{invoker="openshift-internal-ci/release-openshift-origin-installer-launch-azure-modern/1706210448665219072",namespace="openshift-network-operator"} | unpack   |~ "phase" !~ "render" !~ "Render"`)


```
  |   | 2023-09-25 11:03:56 | I0925 09:03:56.085386       1 ovn_kubernetes.go:2430] Upgrade to interconnect, start of phase1: creating IC configmap for single zone |  
-- | -- | -- | -- | --
  |   | 2023-09-25 11:05:45 | W0925 09:05:45.367519       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:06:11 | W0925 09:06:10.992031       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:06:34 | W0925 09:06:34.803227       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:06:58 | W0925 09:06:58.677156       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:07:22 | W0925 09:07:22.606607       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:07:46 | W0925 09:07:46.200657       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:08:10 | W0925 09:08:09.984915       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:08:33 | W0925 09:08:33.786067       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:11:19 | W0925 09:11:19.312712       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:11:41 | W0925 09:11:41.284609       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:12:12 | W0925 09:12:12.886987       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:12:51 | W0925 09:12:51.292337       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:13:13 | W0925 09:13:13.267941       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:13:37 | W0925 09:13:37.059300       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:14:00 | W0925 09:14:00.864583       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:16:24 | W0925 09:16:24.662245       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:16:46 | W0925 09:16:46.642543       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:17:10 | W0925 09:17:10.428839       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:17:34 | W0925 09:17:34.231689       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:17:58 | W0925 09:17:58.039660       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:18:21 | W0925 09:18:21.829188       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:18:45 | W0925 09:18:45.626738       1 ovn_kubernetes.go:2513] Upgrade to interconnect, phase1 is ongoing |  
  |   | 2023-09-25 11:20:10 | I0925 09:20:10.430635       1 ovn_kubernetes.go:2458] Upgrade to interconnect, start of phase2: patching IC configmap for multizone |  
  |   | 2023-09-25 11:20:39 | W0925 09:20:39.662331       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:21:02 | W0925 09:21:01.834686       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:21:31 | W0925 09:21:31.601089       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:21:53 | W0925 09:21:53.783309       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:22:20 | W0925 09:22:20.370987       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:22:42 | W0925 09:22:42.552671       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:23:06 | W0925 09:23:06.566016       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:23:30 | W0925 09:23:30.548944       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:23:55 | W0925 09:23:54.908605       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:24:18 | W0925 09:24:18.544707       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:25:44 | W0925 09:25:44.785257       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:26:07 | W0925 09:26:06.980221       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:26:38 | W0925 09:26:38.421180       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:27:00 | W0925 09:27:00.594873       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:27:26 | W0925 09:27:25.877525       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:27:48 | W0925 09:27:48.602150       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:28:12 | W0925 09:28:12.637789       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:28:36 | W0925 09:28:36.605658       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:29:09 | W0925 09:29:08.970092       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:29:31 | W0925 09:29:31.177213       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:30:00 | W0925 09:30:00.602311       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:31:44 | W0925 09:31:44.295809       1 ovn_kubernetes.go:2516] Upgrade to interconnect, phase2 is ongoing |  
  |   | 2023-09-25 11:32:06 | I0925 09:32:06.673259       1 ovn_kubernetes.go:2491] Upgrade to interconnect, end of phase2: deleting IC configmap, upgrade is done
```